From b9baf2bed6eaf69b4d121e24f78345ada4a426de Mon Sep 17 00:00:00 2001
From: Darshana <darshanasbg@yahoo.com>
Date: Fri, 6 Apr 2012 11:41:21 +0530
Subject: [PATCH] Created display_plug_authentication.lib.php and edited
 server_links.inc.php to add a tab for main tab pane to
 managing authentication plugins.

---
 libraries/display_plug_authentication.lib.php |   55 +++++++++++++++++++++++++
 libraries/server_links.inc.php                |    4 ++
 2 files changed, 59 insertions(+), 0 deletions(-)
 create mode 100644 libraries/display_plug_authentication.lib.php

diff --git a/libraries/display_plug_authentication.lib.php b/libraries/display_plug_authentication.lib.php
new file mode 100644
index 0000000..dfdbbe2
--- /dev/null
+++ b/libraries/display_plug_authentication.lib.php
@@ -0,0 +1,55 @@
+<?php
+/* vim: set expandtab sw=4 ts=4 sts=4: */
+/**
+ *
+ * @package PhpMyAdmin
+ */
+if (! defined('PHPMYADMIN')) {
+    exit;
+}
+
+/**
+ *
+ */
+?>
+<form action="upload_file.php" method="post" enctype="multipart/form-data">
+<div class="exportoptions" id="header">
+    <h2>
+       <?php echo __('Importing & installing authentication plugin to current server'); ?>
+    </h2>
+</div>
+    <div class="importoptions">
+        <h3><?php echo __('Plugin to Import:'); ?></h3>
+    
+            <div class="formelementrow" id="compression_info">
+            Plugin may be Dynamic-link library(dll)or shared object(so).
+            <br />
+            A plugin file name must take the format of <b>[plugin-name].[extention]</b>. Example: <b> sample_plugin.dll</b>
+             <input type="text" name="plugin_na" id="plugin_name" />
+            </div>
+       
+<div class="formelementrow" id="upload_form">
+        <?php PMA_browseUploadFile($max_upload_size); ?>
+        <br />
+</div>
+</div>
+<div class="importoptions">
+        <h3><?php echo __('Plugin:'); ?></h3>
+        <div class="formelementrow">
+            <label ><?php echo __('Plugin name availble in declaration'); ?></label>
+            <input type="text" name="plugin_name" id="plugin_name" />
+        </div>
+        <div class="formelementrow">
+        <label ><?php echo __('Select plugin type'); ?></label>
+        <select name="plugin_type" id="plugin_type">
+                <option value="dll">Dynamic-link library</option>
+                <option value="so">Shared object</option>
+        </select>
+</div>
+</div>
+<input type="text" name="fname" />
+    <div class="importoptions" id="submit">
+        <input type="submit" value="<?php echo __('Go'); ?>" id="buttonGo" multiple="true" />
+    </div>
+ </form>
+
diff --git a/libraries/server_links.inc.php b/libraries/server_links.inc.php
index 489f022..f89a888 100644
--- a/libraries/server_links.inc.php
+++ b/libraries/server_links.inc.php
@@ -108,6 +108,10 @@ if (!$GLOBALS['is_ajax_request']) {
         $tabs['engine']['text'] = __('Engines');
     }
 
+    $tabs['authplugins']['icon'] = 's_plug_auth.png';
+    $tabs['authplugins']['link'] = 'server_plug_authentication.php';
+    $tabs['authplugins']['text'] = __('Plug-Authentication');
+    
     echo PMA_generate_html_tabs($tabs, array());
     unset($tabs);
 
-- 
1.7.9.msysgit.0

